WorkloadManagementSystem/DB/JobDB.py

```diff
@@ -1516,15 +1516,24 @@ def getSiteMaskStatus( self, sites = None ):
     """ Get the currently site mask status
     """
     if isinstance(sites, list):
-      sitesString = ",".join( "'%s'" % site for site in sites)
+      safeSites = []
+      for site in sites:
```

Comment:
Why escaping site by site and not the whole argument ? Isn't there a risk of forging something exactly because it was not escaped as a whole ? (just asking !)

Reply:
The escaping will escape the extra ' characters if it processes the whole thing and they are required so that the SQL server treats it as a list in the final command: 'a','b','c' vs 'a,b,c' (or even '\\'a\\',\\'b\\',\\'c\\'').

Reply:
Ah right, I had not see the single quote !

It makes me think that there is this method: https://github.com/DIRACGrid/DIRAC/blob/integration/Core/Utilities/List.py#L73
But we should escape things here as well ! What do you think ?

Reply:
Hum, unless escaping what is already escaped screws up things...